scripts/serve.ts

```diff
@@ -180,13 +179,22 @@ async function createGateway(name: string) {
     const slug = match[1].toUpperCase().replaceAll('-', '_');
     if (slug in CHAINS) {
       const chain = CHAINS[slug as keyof typeof CHAINS];
-      return new Gateway(
-        new TrustedRollup(
-          createProvider(chain),
-          getProverFactory(chain),
-          new SigningKey(signingKey)
-        )
-      );
+      const provider = createProvider(chain);
+      const key = new SigningKey(signingKey);
+      switch (chain) {
+        case CHAINS.ZKSYNC:
+        case CHAINS.ZKSYNC_SEPOLIA:
+          return new Gateway(new TrustedRollup(provider, ZKSyncProver, key));
+        // NOTE: linea should use eth_getProof instead of linea_getProof
+        // NOTE: this probably needs "--latest" cli option too
+        // rollup => SMT w/Mimc root using linea_getProof
+        // chain => PMT w/Keccak root using eth_getProof
+        // case CHAINS.LINEA:
+        // case CHAINS.LINEA_SEPOLIA:
+        //   return LineaProver;
```

Comment:
These lines should be removed.

Reply:
I think these need to stay for the explanation.  Otherwise, it might seem logical to add them back but the reasoning is pretty nuanced.

Reply:
I agree. It's more that the block doesn't return a Prover directly now, and the former 4 lines still provide the explanatory value.